chain/src/sumtree.rs

```diff
@@ -70,20 +70,17 @@ where
 /// An easy to manipulate structure holding the 3 sum trees necessary to
 /// validate blocks and capturing the UTXO set, the range proofs and the
 /// kernels. Also handles the index of Commitments to positions in the
-/// output and range proof sum trees.
+/// output and range proof pmmr trees.
 ///
 /// Note that the index is never authoritative, only the trees are
 /// guaranteed to indicate whether an output is spent or not. The index
 /// may have commitments that have already been spent, even with
 /// pruning enabled.
-///
-/// In addition of the sumtrees, this maintains the full list of kernel
-/// data so it can be easily packaged for sync or validation.
+
 pub struct SumTrees {
-	output_pmmr_h: PMMRHandle<SumCommit>,
-	rproof_pmmr_h: PMMRHandle<NoSum<RangeProof>>,
-	kernel_pmmr_h: PMMRHandle<NoSum<TxKernel>>,
-	kernel_file: AppendOnlyFile,
+	utxo_pmmr_h: PMMRHandle<OutputStoreable>,
```

Comment:
I'm wondering whether we should call all of these `txo_pmmr*` instead of `utxo_pmmr*`. The reason is that all transaction outputs get stored in it, not only the unspent. They do get pruned when they're spent, which can get confusing but the root is the hash of all TXOs. What do you think? Would that be more or less confusing?

Reply:
There are a lot of things within the code that need renaming at the moment due to this change, so I think I'll open a separate issue where we can decide on these things and then undertake a careful search and replace.

Reply:
:+1: on renaming `utxo_pmmr*` to `txo_pmmr*` (in a separate PR).
I get the confusion now that I have a better understanding of how pruning/compaction works.
Your point that "the root is the hash of all TXOs" just clicked for me (and nicely explains how the Merkle proofs still work even after pruning...)